### PR TITLE
Store reference to debug.traceback on registry

### LINKED ISSUE #3431: Overwriting the global debug variable will break script error handler (DEF-3382)
*Description:*
If a game overwrites the global {{debug}} by accident the internal script error handler (script.cpp, BacktraceErrorHandler) will not work correctly making it output (null)(null) whenever a Lua occurs.

We should either warn if this happen or stop depending on the global variable {{debug}} and use a Lua reference instead.

### PR DESCRIPTION
**Pull request by britzl**
Fixes #3431

### CODE REVIEW

**Review by JCash — DISMISSED** [review]
Looks good!

**Comment by britzl** [comment]
Some tests are failing so I need to look into that before we can merge.

File: engine/script/src/script.cpp [hunk 1]
```diff
@@ -327,6 +326,29 @@ namespace dmScript
...
+        }
+        else if (*entry->what == 'C' || *entry->what == 't')
+        {
+            nwritten = dmSnPrintf(buffer, buffer_size, "  %s:%d: ?\n", entry->short_src, entry->currentline);
+        }
+        else
+        {
+            nwritten = dmSnPrintf(buffer, buffer_size, "  %s:%d: in function <%s:%d>\n", entry->short_src, entry->currentline, entry->short_src, entry->linedefined);
+        }
+        return nwritten;
```
Comment (JCash) [inline engine/script/src/script.cpp:349]: +1

**Review by JCash — APPROVED** [review]
Looks good!

**Pull request merged**